frontend/src/components/db/SearchableDbSourceTree/nodeComponents/TextItemLink.tsx

```diff
@@ -1,59 +1,93 @@
 import React from "react";
 import { NodeApi } from "react-arborist";
+import { useTranslation } from "next-i18next";
 import { currentDbViewAtom } from "@atoms";
+import { Link } from "@components/common/Link";
 import { getTextPath } from "@components/common/utils";
 import { DbSourceTreeNode } from "@components/db/SearchableDbSourceTree/types";
-import { useDbQueryParams } from "@components/hooks/useDbQueryParams";
+import { isDbSourceTreeLeafNodeData } from "@components/db/SearchableDbSourceTree/utils";
+import { useDbRouterParams } from "@components/hooks/useDbRouterParams";
 import { Chip, Tooltip, Typography } from "@mui/material";
 import { useAtomValue } from "jotai";
 
 import { SourceTypeIcon } from "./SourceTypeIcon";
-import { RowBox, TextNameTypography, TextNodeLink } from "./styledComponents";
+import { RowBox, TextNameTypography } from "./styledComponents";
 import styles from "./TextItemLink.module.css";
 
 const CHARACTER_WIDTH = 8;
 const INDENTATION_WIDTH = 90;
 const DEFAULT_NODE_WIDTH = 300;
 
-export function TextItemLink({ node }: { node: NodeApi<DbSourceTreeNode> }) {
-  const { name, fileName, id, dataType } = node.data;
+export function TextItemLinkBody({
+  data,
+  children,
+  treeWidth,
+}: {
+  data: DbSourceTreeNode;
+  children: React.ReactNode;
+  treeWidth: unknown;
```

Comment:
is `unknown` the right type for width?

Reply:
Much of a muchness: `TreeProps<DbSourceTreeNode>.width?: string | number | undefined`. 

`number` is the only type that has an effect.